lib/IDE/CodeCompletion.cpp

```diff
@@ -4025,27 +4038,82 @@ class CompletionOverrideLookup : public swift::VisibleDeclConsumer {
       Builder.addAccessControlKeyword(Access);
   }
 
+  /// Return type if the result type if \p VD should be represented as opaque
+  /// result type.
+  TypeLoc getOpaqueResultTypeLoc(const ValueDecl *VD, DeclVisibilityKind Reason) {
+    if (Reason !=
+        DeclVisibilityKind::MemberOfProtocolImplementedByCurrentNominal)
+      return nullptr;
+
+    auto currTy = CurrDeclContext->getDeclaredTypeInContext();
+    if (!currTy)
+      return nullptr;
+
+    Type ResultT;
+    if (auto *FD = dyn_cast<FuncDecl>(VD))
+      ResultT = FD->getResultInterfaceType();
+    else if (auto *SD = dyn_cast<SubscriptDecl>(VD))
+      ResultT = SD->getElementInterfaceType();
+    else if (auto *VarD = dyn_cast<VarDecl>(VD))
+      ResultT = VarD->getInterfaceType();
+    else
+      return nullptr;
+
+    if (!ResultT->is<DependentMemberType>())
```

Comment:
I'm not sure what this function is doing. Are you printing protocol requirements and extension methods that return associated types as returning an opaque type?

Reply:
Yes, for example:
```swift
protocol P {
  associatedtype Assoc: Collection
  func foo() -> Assoc
}
protocol C: P {
  func // Complete here.
}
```
We wan't to complete `foo()` as `func foo() -> some Collection { <#code#> }`